src/common/Session.cpp

```diff
@@ -151,45 +153,59 @@ namespace SDDM {
 
         qDebug() << "Reading from" << m_fileName;
 
-        QFile file(m_fileName);
-        if (!file.open(QIODevice::ReadOnly))
-            return;
+        QSettings Settings(m_fileName, QSettings::IniFormat);
+        QStringList Locales;
+	Locales << QLocale().name() << QLocale().name().remove(QRegExp(QLatin1String("_.*")));
 
-        QString current_section;
+        if (Settings.status() != QSettings::NoError)
+        	return;
+ 
+        Settings.beginGroup(QLatin1String("Desktop Entry"));
 
-        QTextStream in(&file);
-        while (!in.atEnd()) {
-            QString line = in.readLine();
+        QString Name = Settings.value(QLatin1String("Name"), QLatin1String("")).toString();
+        QString LocalizedName;
 
-            if (line.startsWith(QLatin1String("["))) {
-                // The section name ends before the last ] before the start of a comment
-                int end = line.lastIndexOf(QLatin1Char(']'), line.indexOf(QLatin1Char('#')));
-                if (end != -1)
-                    current_section = line.mid(1, end - 1);
-            }
+        for (QString locale : qAsConst(Locales)) {
+                LocalizedName = Settings.value(QLatin1String("Name") + QLatin1String("[") + locale + QLatin1String("]"), Name).toString();
```

Comment:
This can all be reduced to:

```
        m_displayName = Settings.value(QLatin1String("Name"));
        for (QString locale : qAsConst(Locales)) {
                localizedName = Settings.value(QLatin1String("Name") + QLatin1String("[") + locale + QLatin1String("]")).toString();
                if (!LocalizedName.isEmpty()) {
                        m_displayName = localizedName;
                        break;
                 }
        }
```

Reply:
QLatin1Char for chars.